### DISCUSSION TITLE
Mail-Headers for Message to Subscribers

### DISCUSSION THREAD

**Discussion by pveith** [sympa-community/sympa#1873] [General]
Hi all,
i am responsible for a larger Sympa installation and it works really fine. Thanks for your excellent work!

There is just this one thing, i just seem to not unterstand.
Stituation as i understand it form a header use perspective. Involved headers:
Envelop-To: **subscriber**
To: **list address**
X-Original-To: **list address**

What happens: The Envelop-To is used to determine which account gets the message and then this header gets delete upon delivery (yes, the subscriber address can be found in some Received header, but that is a rather convoluted header; a least in enterprise setups, as we have installed). Thus the normal mailclient user (probably not looking into mail headers at all) can not directly see which of his aliases is subscribed in the list. When they try to unsubscribe for instance, it can fail because they do not use the correct alias for the list. As such we listserver-admin do get tickets, in which they ask for the reason why they can not unsubscribe from a specific list they are subscribed to.

So here is my question:
- Can i configure either the X-Original-To or the To header to actually use the subscriber address? I have not found such a configuration option; i might have overlooked such an option.

I would be inclined to change the To header to subscription address, as sympa generates individual mail for each subscriber. But i don't know the reasoning behind the use of the headers right now.

Best regards,
Patrick Veith

**Comment by ikedas** [comment]
`To:` is a field added by the originator of the message and Sympa does not modify it.

It is undesirable to change this to the (each) address of the final destination. This is because the recipient may mistakenly believe that the message in question is sent only to them, and may not realize that it has been read by the other people on the mailing list.  For the same reason, rewriting `To:` is not appropriate for discussion-type lists: The subscribers cannot find the mailing list address to which the message should be replied.

The feature you desire would be useful in a one-way announcement list, but is difficult to achieve in Sympa. Some other email newsletter distribution systems may allow personalization of `To:`.

**Reply by pveith** [reply]
Thanks for the detailed information. That explains sympas behavior to me.
